modules/file.php

```diff
@@ -340,11 +340,9 @@ function wpcf7_init_uploads() {
 }
 
 function wpcf7_maybe_add_random_dir( $dir ) {
-	do {
-		$rand_max = mt_getrandmax();
-		$rand = zeroise( mt_rand( 0, $rand_max ), strlen( $rand_max ) );
-		$dir_new = path_join( $dir, $rand );
-	} while ( file_exists( $dir_new ) );
+	$rand_max = mt_getrandmax();
+	$rand = zeroise( mt_rand( 0, $rand_max ), strlen( $rand_max ) );
+	$dir_new = path_join( $dir, $rand );
```

Comment:
Could we just change this to use `tempnam( $dir, 'cf7' )` instead?

Reply:
Never mind, apparently it doesn't support stream wrappers 🙄

I still prefer the original approach, since it ensures that there's no conflicts, whereas it's possible for this code to conflict with an existing one. Can we not make S3 respect `file_exists` instead?

Reply:
@rmccue the problem is a dir never exists on S3, so we'd always return `false` which would cause many other issues - however I seem to remember S3 introducing some dir concept at some point which we are not using.

Reply:
@joehoyle I meant more handling the directory checking inside our stream wrapper. I'd rather change it there than in CF7.

Reply:
@rmccue right, that's what I'm talking about. I'm saying it's not possible to handle it in the stream wrapper, because S3 doesn't have directories, therefore in the stream wrapper we always return true for `is_dir` etc

Reply:
Right, but I mean handle it inside `file_exists` instead. Right now, we always return `true` for directories (per my understanding); can we change that?

Reply:
Hmm I might be missing something, but we can only ever have file_exists (well, really is_dir) only ever return true or false, never anything smarter. If we were to change it to _always_ return `false` that would create other issues it's self.